tests/test_signing.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime, timedelta
+
+import itsdangerous
+import pytest
+from pydantic import ValidationError
+
+from hatch import signing
+
+
+def test_token_sign_verify_roundtrip():
+    token1 = signing.AuthToken(
+        url="https://example.com/url",
+        user="user",
+        expiry=datetime.utcnow() + timedelta(minutes=1),
+        scope="view",
+    )
+    token_string = token1.sign()
+
+    token2 = signing.AuthToken.verify(token_string)
+    assert token1 == token2
+
+
+def test_token_url_invalid():
+    with pytest.raises(ValidationError):
+        signing.AuthToken(
+            url="bad",
+            user="user",
+            expiry=datetime.utcnow() + timedelta(minutes=1),
+            scope="bad scope",
```

Comment:
Is it worth setting this to a valid scope so we're only checking the URL?

Reply:
good catch!